src/tests/test_gfd_mining.cpp

```diff
@@ -0,0 +1,118 @@
+#include <cstdlib>
+#include <filesystem>
+#include <fstream>
+#include <vector>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "algorithms/algo_factory.h"
+#include "algorithms/gfd/gfd_miner.h"
+#include "config/names.h"
+#include "csv_config_util.h"
+
+using namespace algos;
+using algos::StdParamsMap;
+
+namespace tests {
+
+class GfdMiningTest : public ::testing::Test {
+public:
+    std::filesystem::path const current_path = kTestDataDir / "graph_data";
+
+    static std::unique_ptr<algos::GfdMiner> CreateGfdMiningInstance(
+            std::filesystem::path const& graph_path, std::size_t const k, std::size_t const sigma) {
+        StdParamsMap option_map = {{config::names::kGraphData, graph_path},
+                                   {config::names::kGfdK, k},
+                                   {config::names::kGfdSigma, sigma}};
+        return algos::CreateAndLoadAlgorithm<GfdMiner>(option_map);
+    }
+};
+
+TEST_F(GfdMiningTest, TestMinGfd) {
+    std::filesystem::path const gfd_path = GfdMiningTest::current_path / "blogs_gfd.dot";
```

Comment:
It would be great to move all the path resolution logic to a separate place, like I did for csv tables.

And here we will use only constants with paths.

Also we should update our current tests for gfd validation